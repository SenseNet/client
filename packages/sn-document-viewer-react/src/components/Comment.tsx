import Avatar from '@material-ui/core/Avatar'
import Button from '@material-ui/core/Button'
import Card from '@material-ui/core/Card'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import CardHeader from '@material-ui/core/CardHeader'
import Collapse from '@material-ui/core/Collapse'
import React, { useState } from 'react'
import { FunctionComponent } from 'react'
import { connect } from 'react-redux'
import styled from 'styled-components'
import { Comment as CommentType } from '../models/Comment'
import { componentType } from '../services'
import { deleteComment, RootReducerType } from '../store'

const mapStateToProps = (state: RootReducerType) => ({
  localization: state.sensenetDocumentViewer.localization,
})

const mapDispatchToProps = {
  deleteComment,
}

const StyledCardContent = styled(CardContent)`
  font-size: 14px;
  letter-spacing: 0.2px;
  line-height: 20px;
  word-wrap: break-word;
`
const DeleteButton = (props: CommentPropType) => (
  <Button size="small" onClick={() => props.deleteComment(props.id)}>
    {props.localization.delete || 'delete'}
  </Button>
)

/**
 * Comment prop type
 */
export type CommentPropType = componentType<typeof mapStateToProps, typeof mapDispatchToProps, CommentType>

const MAX_TEXT_LENGTH = 160

/**
 * Represents a single comment component.
 */
export const CommentComponent: FunctionComponent<CommentPropType> = props => {
  const isLongText = props.text && props.text.length > MAX_TEXT_LENGTH
  const [isOpen, setIsOpen] = useState(!isLongText)
  return (
    <Card>
      <CardHeader
        avatar={<Avatar src={props.createdBy.avatarUrl} alt={props.localization.avatarAlt} />}
        title={props.createdBy.displayName}
      />
      <Collapse in={isOpen} timeout="auto" collapsedHeight={isOpen ? '0px' : '78px'}>
        <StyledCardContent>{props.text}</StyledCardContent>
      </Collapse>
      <CardActions>
        {isLongText ? (
          <>
            <Button size="small" onClick={() => setIsOpen(!isOpen)}>
              {isOpen ? props.localization.showLess || 'Show less' : props.localization.showMore || 'Show more'}
            </Button>
            {isOpen ? <DeleteButton {...props} /> : null}
          </>
        ) : (
          <DeleteButton {...props} />
        )}
      </CardActions>
    </Card>
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CommentComponent)
