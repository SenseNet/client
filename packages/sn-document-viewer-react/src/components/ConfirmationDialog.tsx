import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import React from 'react'
import { FunctionComponent } from 'react'

/**
 * Interface for confirmation dialog component properties
 */
export interface ConfirmationDialogProps {
  dialogTitle: string
  okButtonText?: string
  cancelButtonText?: string
  isOpen: boolean
  onClose: (isCanceled: boolean) => void
}

/**
 * Represents a confirmation dialog
 */
export const ConfirmationDialog: FunctionComponent<ConfirmationDialogProps> = props => {
  const handleCancel = () => props.onClose(true)
  const handleOk = () => props.onClose(false)
  return (
    <Dialog
      disableBackdropClick={true}
      disableEscapeKeyDown={true}
      maxWidth="md"
      aria-labelledby="confirmation-dialog-title"
      aria-describedby="confirmation-dialog-description"
      open={props.isOpen}>
      <DialogTitle id="confirmation-dialog-title">{props.dialogTitle}</DialogTitle>
      <DialogContent>
        <DialogContentText
          dangerouslySetInnerHTML={{ __html: props.children as string }}
          id="confirmation-dialog-description"
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={handleOk} variant="contained" color="primary">
          {props.okButtonText || 'ok'}
        </Button>
        <Button onClick={handleCancel}>{props.cancelButtonText || 'cancel'}</Button>
      </DialogActions>
    </Dialog>
  )
}
