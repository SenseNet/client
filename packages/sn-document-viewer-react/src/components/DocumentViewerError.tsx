import React from 'react'
import { Typography } from '@material-ui/core'
import { useDocumentData, useLocalization } from '../hooks'
import { LayoutAppBar } from '.'

/**
 * Component to display viewer related errors
 */
export const DocumentViewerError: React.FC = () => {
  const { documentData } = useDocumentData()
  const localization = useLocalization()

  return (
    <div
      style={{
        display: 'flex',
        justifyContent: 'space-around',
        alignContent: 'center',
        alignItems: 'center',
        height: '100%',
      }}>
      <LayoutAppBar style={{ position: 'fixed', top: 0 }}>
        <span />
      </LayoutAppBar>
      <div
        style={{
          display: 'flex',
          justifyContent: 'space-between',
          alignItems: 'center',
          flexDirection: 'column',
          maxWidth: 500,
          margin: '.5em 0 .6em 0',
        }}>
        <svg width="442px" height="290px" viewBox="0 0 442 240" version="1.1" xmlns="http://www.w3.org/2000/svg">
          <g id="Page-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
            <g id="No-provider" transform="translate(-9.000000, 5.000000)">
              <g id="Document" transform="translate(9.000000, 0.000000)">
                <ellipse id="Oval-2" fill="#DFDFDF" cx="221" cy="230" rx="221" ry="27" />
                <path
                  d="M129.868875,229.165007 L129.868875,6.81087018 C129.868875,3.49716168 132.555166,0.810870179 135.868875,0.810870179 L262.245476,0.810870179 C263.836775,0.810870179 265.362898,1.44301122 266.488117,2.56822949 L311.211391,47.2915035 C312.336609,48.4167218 312.96875,49.9428453 312.96875,51.5341442 L312.96875,229.724935 C312.96875,233.038644 310.282458,235.724935 306.96875,235.724935 C306.962205,235.724935 306.95566,235.724924 306.949115,235.724903 L135.84924,235.164975 C132.543214,235.154156 129.868875,232.47105 129.868875,229.165007 Z"
                  id="Path"
                  stroke="#A4A8AD"
                  strokeWidth="11"
                  fill="#EEEEEE"
                />
                <ellipse id="Oval" fill="#A4A8AD" cx="184" cy="120.5" rx="9" ry="10.5" />
                <ellipse id="Oval-Copy" fill="#A4A8AD" cx="259" cy="120.5" rx="9" ry="10.5" />
                <path
                  d="M203.877833,155.306526 C215.78965,143.944752 227.408534,143.944752 238.734486,155.306526"
                  id="Path-2"
                  stroke="#A4A8AD"
                  strokeWidth="7"
                  strokeLinecap="round"
                />
                <path
                  d="M262.50809,3.20349862 L262.50809,46.2114256 C262.50809,47.3159951 263.40352,48.2114256 264.50809,48.2114256 L308.124241,48.2114256"
                  id="Path-2"
                  stroke="#A4A8AD"
                  strokeWidth="11"
                />
              </g>
            </g>
          </g>
        </svg>
        <Typography variant="h5" color="textSecondary" align="center" style={{ fontWeight: 'bolder' }}>
          {localization.errorLoadingDocument.find((e) => e.code === documentData.pageCount)}
        </Typography>
        <Typography variant="subtitle1" color="textSecondary" align="center" style={{ whiteSpace: 'pre-wrap' }}>
          {localization.errorLoadingDetails}
        </Typography>
      </div>
    </div>
  )
}
