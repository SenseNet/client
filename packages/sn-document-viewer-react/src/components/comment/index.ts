export * from './comment'
export * from './create-comment'
export * from './delete-comment-button'
export * from './style'
