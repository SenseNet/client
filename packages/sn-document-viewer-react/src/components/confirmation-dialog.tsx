import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core'
import React from 'react'

/**
 * Interface for confirmation dialog component properties
 */
export interface ConfirmationDialogProps {
  dialogTitle: string
  okButtonText?: string
  cancelButtonText?: string
  isOpen: boolean
  onClose: (isCanceled: boolean) => void
}

/**
 * Represents a confirmation dialog
 */
export const ConfirmationDialog: React.FC<ConfirmationDialogProps> = (props) => {
  const handleCancel = () => props.onClose(true)
  const handleOk = () => props.onClose(false)
  return (
    <Dialog
      disableBackdropClick={true}
      disableEscapeKeyDown={true}
      maxWidth="md"
      aria-labelledby="confirmation-dialog-title"
      aria-describedby="confirmation-dialog-description"
      open={props.isOpen}>
      <DialogTitle id="confirmation-dialog-title">{props.dialogTitle}</DialogTitle>
      <DialogContent>
        <DialogContentText
          dangerouslySetInnerHTML={{ __html: props.children as string }}
          id="confirmation-dialog-description"
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={handleOk} variant="contained" color="primary">
          {props.okButtonText || 'ok'}
        </Button>
        <Button onClick={handleCancel}>{props.cancelButtonText || 'cancel'}</Button>
      </DialogActions>
    </Dialog>
  )
}
