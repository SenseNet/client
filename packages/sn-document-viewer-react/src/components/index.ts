export * from './comment'
export * from './document-widgets'
export * from './page-widgets'
export * from './ConfirmationDialog'
export * from './DocumentViewer'
export * from './DocumentViewerError'
export * from './DocumentViewerLayout'
export * from './DocumentViewerLoading'
export * from './DocumentViewerRegeneratePreviews'
export * from './LayoutAppBar'
export * from './Page'
export * from './PageList'
export * from './styles'
