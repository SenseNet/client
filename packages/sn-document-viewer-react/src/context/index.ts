export * from './api-settings'
export * from './comment-states'
export * from './comments'
export * from './document-data'
export * from './document-permissions'
export * from './localization-context'
export * from './preview-image-data'
export * from './viewer-settings'
export * from './viewer-state'
