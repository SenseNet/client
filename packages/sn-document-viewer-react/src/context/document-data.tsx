import { DocumentData } from '@sensenet/client-core'
import { deepMerge, DeepPartial, sleepAsync } from '@sensenet/client-utils'
import { useRepository } from '@sensenet/hooks-react'
import React, { useCallback, useEffect, useState } from 'react'
import Semaphore from 'semaphore-async-await'
import { POLLING_INTERVAL } from '../components'
import { PreviewState } from '../enums'
import { useDocumentViewerApi, useViewerSettings } from '../hooks'

const defaultDocumentData: DocumentData = {
  documentName: '',
  documentType: '',
  fileSizekB: 0,
  hostName: '',
  idOrPath: 0,
  pageCount: PreviewState.Loading,
  shapes: {
    annotations: [],
    highlights: [],
    redactions: [],
  },
  error: undefined,
}

export interface DocumentDataContextType {
  documentData: DocumentData
  updateDocumentData: (data: DeepPartial<DocumentData>) => void
  isInProgress: boolean
  triggerReload: () => void
}

export const defaultDocumentDataContextValue: DocumentDataContextType = {
  documentData: defaultDocumentData,
  updateDocumentData: () => undefined,
  isInProgress: false,
  triggerReload: () => {},
}

export const DocumentDataContext = React.createContext<DocumentDataContextType>(defaultDocumentDataContextValue)

export const DocumentDataProvider: React.FC = ({ children }) => {
  const api = useDocumentViewerApi()
  const doc = useViewerSettings()
  const repo = useRepository()

  const [documentData, setDocumentData] = useState<DocumentData>(defaultDocumentData)
  const [loadLock] = useState(new Semaphore(1))
  const [isInProgress, setIsInProgress] = useState(false)
  const [reloadToken, setReloadToken] = useState(0)

  const triggerReload = () => setReloadToken((prevValue) => prevValue + 1)

  useEffect(() => {
    const ac = new AbortController()
    const getData = async () => {
      const result = await api.getDocumentData({
        hostName: repo.configuration.repositoryUrl,
        idOrPath: doc.documentIdOrPath,
        version: doc.version,
        abortController: ac,
      })
      if (result.pageCount === PreviewState.Loading && ac.signal.aborted === false) {
        setDocumentData(result)
        await sleepAsync(POLLING_INTERVAL)
        getData()
      } else {
        setDocumentData(result)
      }
    }
    ;(async () => {
      try {
        setIsInProgress(true)
        await loadLock.acquire()
        await getData()
      } catch (error) {
        if (!ac.signal.aborted) {
          setDocumentData({ ...defaultDocumentData, pageCount: PreviewState.ClientFailure, error: error.toString() })
          throw error
        }
      } finally {
        setIsInProgress(false)
        loadLock.release()
      }
    })()
    return () => ac.abort()
  }, [api, doc.documentIdOrPath, doc.version, loadLock, repo.configuration.repositoryUrl, reloadToken])

  const updateDocumentData = useCallback(
    (newDocData: Partial<DocumentData>) => {
      const merged = deepMerge(documentData, newDocData)
      if (JSON.stringify(documentData) !== JSON.stringify(merged)) {
        setDocumentData(merged)
      }
    },
    [documentData],
  )

  if (!documentData.idOrPath) {
    return null
  }

  return (
    <DocumentDataContext.Provider value={{ documentData, updateDocumentData, isInProgress, triggerReload }}>
      {children}
    </DocumentDataContext.Provider>
  )
}
