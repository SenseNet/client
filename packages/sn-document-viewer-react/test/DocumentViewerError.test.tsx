import { mount } from 'enzyme'
import React from 'react'
import { DocumentViewerError } from '../src/components/DocumentViewerError'
import { DocumentDataContext } from '../src/context/document-data'
import { PreviewState } from '../src/Enums'
import { exampleDocumentData } from './__Mocks__/viewercontext'

describe('Document Viewer Error component', () => {
  it('matches snapshot wihtout provided context', () => {
    const wrapper = mount(<DocumentViewerError />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot with provided context', () => {
    const wrapper = mount(
      <DocumentDataContext.Provider
        value={{
          documentData: { ...exampleDocumentData, pageCount: PreviewState.UploadFailure, error: ':(' },
          updateDocumentData: async () => undefined,
        }}>
        <DocumentViewerError />
      </DocumentDataContext.Provider>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
