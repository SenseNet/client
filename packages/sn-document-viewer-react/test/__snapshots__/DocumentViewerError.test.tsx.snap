// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Viewer Error component matches snapshot with provided context 1`] = `
<DocumentViewerError>
  <div
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "space-around",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "margin": ".5em 0 .6em 0",
          "maxWidth": 500,
        }
      }
    >
      <svg
        height="290px"
        version="1.1"
        viewBox="0 0 442 240"
        width="442px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          id="Page-1"
          stroke="none"
          strokeWidth="1"
        >
          <g
            id="No-provider"
            transform="translate(-9.000000, 5.000000)"
          >
            <g
              id="Document"
              transform="translate(9.000000, 0.000000)"
            >
              <ellipse
                cx="221"
                cy="230"
                fill="#DFDFDF"
                id="Oval-2"
                rx="221"
                ry="27"
              />
              <path
                d="M129.868875,229.165007 L129.868875,6.81087018 C129.868875,3.49716168 132.555166,0.810870179 135.868875,0.810870179 L262.245476,0.810870179 C263.836775,0.810870179 265.362898,1.44301122 266.488117,2.56822949 L311.211391,47.2915035 C312.336609,48.4167218 312.96875,49.9428453 312.96875,51.5341442 L312.96875,229.724935 C312.96875,233.038644 310.282458,235.724935 306.96875,235.724935 C306.962205,235.724935 306.95566,235.724924 306.949115,235.724903 L135.84924,235.164975 C132.543214,235.154156 129.868875,232.47105 129.868875,229.165007 Z"
                fill="#EEEEEE"
                id="Path"
                stroke="#A4A8AD"
                strokeWidth="11"
              />
              <ellipse
                cx="184"
                cy="120.5"
                fill="#A4A8AD"
                id="Oval"
                rx="9"
                ry="10.5"
              />
              <ellipse
                cx="259"
                cy="120.5"
                fill="#A4A8AD"
                id="Oval-Copy"
                rx="9"
                ry="10.5"
              />
              <path
                d="M203.877833,155.306526 C215.78965,143.944752 227.408534,143.944752 238.734486,155.306526"
                id="Path-2"
                stroke="#A4A8AD"
                strokeLinecap="round"
                strokeWidth="7"
              />
              <path
                d="M262.50809,3.20349862 L262.50809,46.2114256 C262.50809,47.3159951 263.40352,48.2114256 264.50809,48.2114256 L308.124241,48.2114256"
                id="Path-2"
                stroke="#A4A8AD"
                strokeWidth="11"
              />
            </g>
          </g>
        </g>
      </svg>
      <WithStyles(ForwardRef(Typography))
        align="center"
        color="textSecondary"
        style={
          Object {
            "whiteSpace": "pre-wrap",
          }
        }
        variant="subtitle1"
      >
        <ForwardRef(Typography)
          align="center"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          color="textSecondary"
          style={
            Object {
              "whiteSpace": "pre-wrap",
            }
          }
          variant="subtitle1"
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            style={
              Object {
                "whiteSpace": "pre-wrap",
              }
            }
          >
            The following error occured during opening a document: 
          </h6>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        align="center"
        color="textSecondary"
        style={
          Object {
            "fontWeight": "bolder",
          }
        }
        variant="h5"
      >
        <ForwardRef(Typography)
          align="center"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          color="textSecondary"
          style={
            Object {
              "fontWeight": "bolder",
            }
          }
          variant="h5"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            style={
              Object {
                "fontWeight": "bolder",
              }
            }
          >
            Error during preview generation! No preview available.
          </h5>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
    </div>
  </div>
</DocumentViewerError>
`;

exports[`Document Viewer Error component matches snapshot without provided context 1`] = `
<DocumentViewerError>
  <div
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "space-around",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "margin": ".5em 0 .6em 0",
          "maxWidth": 500,
        }
      }
    >
      <svg
        height="290px"
        version="1.1"
        viewBox="0 0 442 240"
        width="442px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          id="Page-1"
          stroke="none"
          strokeWidth="1"
        >
          <g
            id="No-provider"
            transform="translate(-9.000000, 5.000000)"
          >
            <g
              id="Document"
              transform="translate(9.000000, 0.000000)"
            >
              <ellipse
                cx="221"
                cy="230"
                fill="#DFDFDF"
                id="Oval-2"
                rx="221"
                ry="27"
              />
              <path
                d="M129.868875,229.165007 L129.868875,6.81087018 C129.868875,3.49716168 132.555166,0.810870179 135.868875,0.810870179 L262.245476,0.810870179 C263.836775,0.810870179 265.362898,1.44301122 266.488117,2.56822949 L311.211391,47.2915035 C312.336609,48.4167218 312.96875,49.9428453 312.96875,51.5341442 L312.96875,229.724935 C312.96875,233.038644 310.282458,235.724935 306.96875,235.724935 C306.962205,235.724935 306.95566,235.724924 306.949115,235.724903 L135.84924,235.164975 C132.543214,235.154156 129.868875,232.47105 129.868875,229.165007 Z"
                fill="#EEEEEE"
                id="Path"
                stroke="#A4A8AD"
                strokeWidth="11"
              />
              <ellipse
                cx="184"
                cy="120.5"
                fill="#A4A8AD"
                id="Oval"
                rx="9"
                ry="10.5"
              />
              <ellipse
                cx="259"
                cy="120.5"
                fill="#A4A8AD"
                id="Oval-Copy"
                rx="9"
                ry="10.5"
              />
              <path
                d="M203.877833,155.306526 C215.78965,143.944752 227.408534,143.944752 238.734486,155.306526"
                id="Path-2"
                stroke="#A4A8AD"
                strokeLinecap="round"
                strokeWidth="7"
              />
              <path
                d="M262.50809,3.20349862 L262.50809,46.2114256 C262.50809,47.3159951 263.40352,48.2114256 264.50809,48.2114256 L308.124241,48.2114256"
                id="Path-2"
                stroke="#A4A8AD"
                strokeWidth="11"
              />
            </g>
          </g>
        </g>
      </svg>
      <WithStyles(ForwardRef(Typography))
        align="center"
        color="textSecondary"
        style={
          Object {
            "whiteSpace": "pre-wrap",
          }
        }
        variant="subtitle1"
      >
        <ForwardRef(Typography)
          align="center"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          color="textSecondary"
          style={
            Object {
              "whiteSpace": "pre-wrap",
            }
          }
          variant="subtitle1"
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            style={
              Object {
                "whiteSpace": "pre-wrap",
              }
            }
          >
            The following error occured during opening a document: 
          </h6>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        align="center"
        color="textSecondary"
        style={
          Object {
            "fontWeight": "bolder",
          }
        }
        variant="h5"
      >
        <ForwardRef(Typography)
          align="center"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          color="textSecondary"
          style={
            Object {
              "fontWeight": "bolder",
            }
          }
          variant="h5"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            style={
              Object {
                "fontWeight": "bolder",
              }
            }
          >
            Unknown error
          </h5>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
    </div>
  </div>
</DocumentViewerError>
`;
